app-android/src/main/java/com/ustadmobile/port/android/view/ScopedGrantEditFragment.kt

```diff
@@ -87,4 +100,55 @@ class ScopedGrantEditFragment: UstadEditFragment<ScopedGrant>(), ScopedGrantEdit
             field = value
             mBinding?.fieldsEnabled = value
         }
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun ScopedGrantEditScreen(
+    uiState: ScopedGrantEditUiState = ScopedGrantEditUiState(),
+    onChangedBitmask: (BitmaskFlag?) -> Unit = {},
+) {
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+    ) {
+        uiState.bitmaskList.forEach { bitmask ->
+
+            ListItem(
```

Comment:
Using the switch inside a listitem like this is OK and is a valid use case as per the documentation

ListItem here needs to use modifier.toggleable. Using the current version, you can only tap on the switch itself. The user should also be able to tap anywhere on the text to toggle the switch.

e.g.

```
ListItem(
  modifier = modifier.toggleable( 
     role = Role.Switch,
     value = bitmask.enabled,
     onValueChange = {
         onBitmaskChanged(bitmask.copy(enabled = it))
     }
  ),
  text = { Text(messageIdResource(...) },
  trailing =  {
        Switch(
            //Note: do not use the event handler on Switch. It is handled by the toggleable on container
            checked = bitmask.enabled,
        )
    }
)
```